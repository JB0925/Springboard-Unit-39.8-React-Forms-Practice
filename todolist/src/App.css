.html {
  box-sizing: border-box;
  font-family: 'Roboto', sans-serif;
}

*, *::before, *::after {
  box-sizing: inherit;
}

body {
  margin: 0;
  padding: 0;
}

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
}

.App-header h1 {
  text-align: center;
  font-size: 1.6rem;
}

