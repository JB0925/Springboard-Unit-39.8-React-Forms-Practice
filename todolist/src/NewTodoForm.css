form {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
    padding: 25px;
    border: .5px solid lightgray;
    border-radius: 5px;
}

input, label, button {
    margin-top: 10px;
}

input, button {
    height: 35px;
    width: 80%;
}

button {
    background-color: dodgerblue;
    color: white;
    width: 78%;
}