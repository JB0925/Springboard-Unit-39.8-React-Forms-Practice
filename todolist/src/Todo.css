.Todo {
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: left;
    width: 100%;
}

.Todo-message {
    margin-left: 20%;
}

.Todo-button {
    margin-left: auto;
    margin-right: 10%;
}