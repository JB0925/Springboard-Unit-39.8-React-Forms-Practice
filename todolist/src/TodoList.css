.TodoList {
    display: flex;
    align-items: center;
    justify-content: space-around;
    border: 1px solid lightgray;
    border-radius: 5px;
    width: 55%;
    /* padding: 25px; */
}

.TodoList-form {
    width: 50%;
}

.TodoList-todos {
    width: 50%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.TodoList-todos button {
    padding: 1px;
    margin-left: 25px;
    text-align: center;
}

@media screen and (max-width: 900px) {
    .TodoList {
        flex-direction: column;
        padding: 10px;
    }

    .TodoList-form, .TodoList-todos {
        width: 100%;
    }

    form {
        width: auto;
    }

    .Todo-message {
        margin-left: 5%;
    }
}