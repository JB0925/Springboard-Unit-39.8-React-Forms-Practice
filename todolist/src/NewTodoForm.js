import React, { useState } from "react";
import "NewTodoForm.css";
