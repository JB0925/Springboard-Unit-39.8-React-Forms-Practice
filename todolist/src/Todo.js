import React, { useState } from "react";
import "./Todo.css";

const Todo = ({ message, removeTodo }) => {
    message = message === "" ? "No Message Provided" : message;

    const handleClick = () => {
        removeTodo(message);
    };

    return (
        <div className="Todo">
            <div className="Todo-message">
                {message}
            </div>
            <div className="Todo-button">
                <button onClick={handleClick}>X</button>
            </div>
        </div>
    );
};

export default Todo;