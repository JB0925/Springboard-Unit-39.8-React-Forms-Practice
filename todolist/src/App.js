import TodoList from './TodoList';
import './App.css';

function App() {
  return (
    <div className="App">
      <div className="App-header">
          <h1>Your ToDo List</h1>
      </div>
      <TodoList />
    </div>
  );
}

export default App;
