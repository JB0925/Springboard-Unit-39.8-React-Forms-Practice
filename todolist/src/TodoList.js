import React, { useState } from "react";
import Todo from "./Todo";
import "./TodoList.css";
import {v4 as uuid} from "uuid";

const TodoList = () => {
    const [todos, updateTodos] = useState([]);
    const addTodo = newTodo => {
        updateTodos([
            ...todos,
            newTodo
        ]);
    };

    const removeTodo = oldTodo => {
        updateTodos(() => {
            const filteredTodos = todos.filter(todo => todo !== oldTodo);
            return filteredTodos;
        });
    };

    const allTodos = todos.map(todo => <Todo 
                                        message={todo} 
                                        removeTodo={removeTodo} 
                                        key={uuid()}/>);

    return (
        <div className="TodoList">
            <div className="TodoList-header">
                <h1>Your ToDo List</h1>
            </div>
            <div className="TodoList-todos">
                {allTodos}
            </div>
        </div>
    );
};

export default TodoList;