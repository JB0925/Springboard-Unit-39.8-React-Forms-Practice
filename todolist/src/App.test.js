import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import TodoList from "./TodoList";

test('The App component will render.', () => {
  render(<App />);
});

test('The NewTodoForm component will render.', () => {
  render(<NewTodoForm />);
});

test('The Todo component will render.', () => {
  render(<Todo />);
});

test('The TodoList component will render.', () => {
  render(<TodoList />);
});

test('The App component matches the snapshot.', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test('The NewTodoForm component matches the snapshot.', () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

test('The Todo component matches the snapshot.', () => {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

test('The TodoList component matches the snapshot.', () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

test('The NewTodoForm renders a new Todo component in the DOM, and the remove button removes it.', () => {
  const { getByText, queryByText, getByLabelText } = render(<TodoList />);
  const input = getByLabelText("Enter a New Todo");
  const btn = getByText('Add Todo');

  expect(document.body).not.toContainHTML('class="Todo"');

  fireEvent.change(input, { target: { value: 'mow the grass' }});
  fireEvent.click(btn);

  expect(document.body).toContainHTML('class="Todo');

  const removeBtn = getByText('X');
  fireEvent.click(removeBtn);
  expect(document.body).not.toContainHTML('class="Todo"');
});
