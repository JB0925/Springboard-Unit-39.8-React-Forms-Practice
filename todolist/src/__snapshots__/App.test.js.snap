// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App component matches the snapshot. 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="App-header"
    >
      <h1>
        Your ToDo List
      </h1>
    </div>
    <div
      class="TodoList"
    >
      <div
        class="TodoList-form"
      >
        <form>
          <label
            for="newTodo"
          >
            Enter a New Todo
          </label>
          <input
            id="newTodo"
            name="newTodo"
            placeholder="i.e. Mow The Grass"
            type="text"
            value=""
          />
          <button
            type="submit"
          >
            Add Todo
          </button>
        </form>
      </div>
      <div
        class="TodoList-todos"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`The NewTodoForm component matches the snapshot. 1`] = `
<DocumentFragment>
  <form>
    <label
      for="newTodo"
    >
      Enter a New Todo
    </label>
    <input
      id="newTodo"
      name="newTodo"
      placeholder="i.e. Mow The Grass"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;

exports[`The Todo component matches the snapshot. 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <div
      class="Todo-message"
    />
    <div
      class="Todo-button"
    >
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`The TodoList component matches the snapshot. 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <div
      class="TodoList-form"
    >
      <form>
        <label
          for="newTodo"
        >
          Enter a New Todo
        </label>
        <input
          id="newTodo"
          name="newTodo"
          placeholder="i.e. Mow The Grass"
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Add Todo
        </button>
      </form>
    </div>
    <div
      class="TodoList-todos"
    />
  </div>
</DocumentFragment>
`;
